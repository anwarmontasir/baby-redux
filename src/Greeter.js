import React from 'react';
import { connect } from 'react-redux';
import { updateSalutation, updateName, reset } from './greeter.actions';

export function Greeter({ name, salutation, dispatch }) {
  return (
    <div>
      <span>{salutation}</span>
      <span>{name}</span>!

      <form>
        Salutation: <input onChange={event => dispatch(updateSalutation(event.target.value))} />
        Name: <input onChange={event => dispatch(updateName(event.target.value))} />
        <button onClick={() => dispatch(reset())}>Reset</button>
      </form>
    </div>
  );
}

export default connect(
  state => ({ 
    salutation: state.salutation,
    name: state.name
  })
)(Greeter);