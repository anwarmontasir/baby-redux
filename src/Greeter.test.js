import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Greeter } from './Greeter';

describe('Greeter', () => {

  it('updates salutation and name', () => {
    const wrapper = shallow(<Greeter salutation='Howdy' name='Pardner' />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('resets salutation and name', () => {
    const wrapper = shallow(<Greeter salutation='Hello' name='Stranger' />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});