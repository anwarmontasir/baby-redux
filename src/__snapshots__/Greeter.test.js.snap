// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter resets salutation and name 1`] = `
<div>
  <span>
    Hello
     
  </span>
  <span>
    Stranger
  </span>
  !
  <form>
    <label
      for="salutation"
    >
      Salutation:
    </label>
     
    <input
      id="salutation"
      onChange={[Function]}
    />
    <label
      for="name"
    >
      Name:
    </label>
     
    <input
      id="name"
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Greeter updates salutation and name 1`] = `
<div>
  <span>
    Howdy
     
  </span>
  <span>
    Pardner
  </span>
  !
  <form>
    <label
      for="salutation"
    >
      Salutation:
    </label>
     
    <input
      id="salutation"
      onChange={[Function]}
    />
    <label
      for="name"
    >
      Name:
    </label>
     
    <input
      id="name"
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </form>
</div>
`;
